workflow-service/src/main/java/com/redhat/parodos/workflow/definition/repository/WorkFlowTaskDefinitionRepository.java

```diff
@@ -29,4 +31,6 @@ public interface WorkFlowTaskDefinitionRepository extends JpaRepository<WorkFlow
 
 	WorkFlowTaskDefinition findFirstByName(String name);
 
+	WorkFlowTaskDefinition findFirstByNameAndWorkFlowDefinitionType(String name, String workFlowType);
```

Comment:
Why `String workFlowType` instead of `WorkFlowType` obj ?

Reply:
The type (workFlowType) was defined as String in the entity. 
Charge done to enforce the type.